--- level.cpp.orig	2010-04-27 14:23:54.000000000 +0000
+++ level.cpp	2010-04-27 14:27:16.000000000 +0000
@@ -30,44 +30,44 @@
 IMPLEMENT_LOG4CXX_OBJECT_WITH_CUSTOM_CLASS(Level, LevelClass)
 
 LevelPtr Level::getOff() {
-   static LevelPtr level(new Level(Level::OFF_INT, LOG4CXX_STR("OFF"), 0));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::OFF_INT, LOG4CXX_STR("OFF"), 0));
+   return *level;
 }
 
 LevelPtr Level::getFatal() {
-   static LevelPtr level(new Level(Level::FATAL_INT, LOG4CXX_STR("FATAL"), 0));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::FATAL_INT, LOG4CXX_STR("FATAL"), 0));
+   return *level;
 }
 
 LevelPtr Level::getError() {
-   static LevelPtr level(new Level(Level::ERROR_INT, LOG4CXX_STR("ERROR"), 3));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::ERROR_INT, LOG4CXX_STR("ERROR"), 3));
+   return *level;
 }
 
 LevelPtr Level::getWarn() {
-   static LevelPtr level(new Level(Level::WARN_INT, LOG4CXX_STR("WARN"), 4));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::WARN_INT, LOG4CXX_STR("WARN"), 4));
+   return *level;
 }
 
 LevelPtr Level::getInfo() {
-   static LevelPtr level(new Level(Level::INFO_INT, LOG4CXX_STR("INFO"), 6));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::INFO_INT, LOG4CXX_STR("INFO"), 6));
+   return *level;
 }
 
 LevelPtr Level::getDebug() {
-   static LevelPtr level(new Level(Level::DEBUG_INT, LOG4CXX_STR("DEBUG"), 7));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::DEBUG_INT, LOG4CXX_STR("DEBUG"), 7));
+   return *level;
 }
 
 LevelPtr Level::getTrace() {
-   static LevelPtr level(new Level(Level::TRACE_INT, LOG4CXX_STR("TRACE"), 7));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::TRACE_INT, LOG4CXX_STR("TRACE"), 7));
+   return *level;
 }
 
 
 LevelPtr Level::getAll() {
-   static LevelPtr level(new Level(Level::ALL_INT, LOG4CXX_STR("ALL"), 7));
-   return level;
+   static LevelPtr *level=new LevelPtr(new Level(Level::ALL_INT, LOG4CXX_STR("ALL"), 7));
+   return *level;
 }
 
 
